modules/engine-prototype/src/main/scala/engine/Step.scala

```diff
@@ -0,0 +1,251 @@
+package engine
+
+import scala.concurrent.ExecutionContext
+
+import cats.implicits._
+import cats.effect.Effect
+
+sealed trait Step
+object Step {
+
+  // Whatever is shared among all Steps. Notice it doesn't // extend Step
+  // because it's meant to be used only internally
+  case class Core(
+    id: String,
+    config: Core.Configuration,
+    breakpoint: Boolean
+  )
+
+  object Core {
+    case class Configuration()
+  }
+
+  sealed trait F2 extends Step
+  object F2 {
+
+    case class Done(core: Core, fileId: String) extends F2
+
+    sealed trait Pending extends F2
+    object Pending {
+      case class Standard(core: Core) extends Pending
+      case class Flat(core: Core) extends Pending
+      case class Dark(core: Core) extends Pending
+    }
+
+    sealed trait Ongoing extends F2
+    object Ongoing {
+      final case class Standard(core: Core, tcsConfigured: Boolean, instConfigured: Boolean) extends Ongoing
```

Comment:
These are just dummy parameters to make a point, but we could put anything there to represent a Step under execution. For example, an estimated timestamp of completion for progress bars that could be updated if the estimation taken from the EPICS channels changes.